src/main/java/org/example/controller/ProfessorController.java

```diff
@@ -0,0 +1,66 @@
+package org.example.controller;
+
+import org.example.entities.Professor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.example.service.ProfessorService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/professors")
```

Comment:
you can add /api at request mapping ... so we will have /api/professor/{id} or /api/professor